ambari-server/src/main/python/ambari_server/serverSetup.py

```diff
@@ -1292,7 +1303,7 @@ def check_ambari_java_version_is_valid(java_home, java_bin, min_version, propert
       err = "Checking JDK version command returned with exit code %s" % process.returncode
       raise FatalException(process.returncode, err)
     else:
-      actual_jdk_version = int(out)
+      actual_jdk_version = int(get_java_major_version(out))
```

Comment:
I think `STACK_JAVA_VERSION` should be set to `actual_jdk_version` instead of `out`, since `out` now has more content than just the major version number:

https://github.com/apache/ambari/blob/44436e1d5e0d28e07285c7b506cf596748ce85fc/ambari-server/src/main/python/ambari_server/serverSetup.py#L1299